src/baseline.rs

```diff
@@ -242,6 +243,43 @@ fn create_rustdoc_manifest_for_crate_version(
     }
 }
 
+fn choose_baseline_version(
+    crate_: &Crate,
+    version_current: Option<&semver::Version>,
+) -> anyhow::Result<String> {
+    if let Some(current) = version_current {
+        let mut instances = crate_
+            .versions()
+            .iter()
+            .filter(|i| !i.is_yanked())
+            .filter_map(|i| semver::Version::parse(i.version()).ok())
+            // For unpublished changes when the user doesn't increment the version
+            // post-release, allow using the current version as a baseline.
+            .filter(|v| v <= current)
+            .collect::<Vec<_>>();
+        instances.sort();
+        instances
+            .iter()
+            .rev()
+            .find(|v| v.pre.is_empty())
+            .or_else(|| instances.last())
+            .map(|v| v.to_string())
+            .with_context(|| {
+                anyhow::format_err!(
+                    "No available baseline versions for {}@{}",
+                    crate_.name(),
+                    current
+                )
+            })
+    } else {
+        let instance = crate_
+            .highest_normal_version()
+            .unwrap_or_else(|| crate_.highest_version())
```

Comment:
I know that this code is unchanged in this PR, but I don't really understand this line -- maybe it could be commented to explain what's going on here @obi1kenobi. What is considered a "normal" version and why it is OK to choose a "not normal" (whatever is "normal") version when there isn't a normal one? Assuming that a "normal" possibly means non-yanked and non-pre-release, what would happen when there are only "not normal" versions?

Reply:
Your guess was correct, it's "not yanked, not pre-release": https://docs.rs/crates-index/latest/crates_index/struct.Crate.html#method.highest_normal_version

If there's no "normal" version, then our options are:
- error
- choose the best option from what's available

The choice isn't necessarily obvious, and this isn't my code. But I think the decision to try to "guess what the user wants" in this case rather than making it a hard error is a reasonable one: in the worst case, we error anyway, and in the best case we guess correctly and delight the user.

Leaving a comment is reasonable. It might just restate what the docs for the two methods on `Crate` already say, but that's alright if it saves the next reader of the code a Google search.

Reply:
Can you leave a comment here (in this code) in this PR @mgr0dzicki ?

Reply:
Done! Please have a look at it @tonowak